mtags/src/main/scala-2/scala/meta/internal/mtags/ScalaMtags.scala

```diff
@@ -160,20 +202,146 @@ class ScalaMtags(val input: Input.VirtualFile)
         case t: Defn.Object =>
           term(t.name, Kind.OBJECT, 0); continue()
         case t: Defn.Type =>
-          tpe(t.name, Kind.TYPE, 0); stop()
-          enterTypeParameters(t.tparams)
+          withScala3Owner {
+            tpe(t.name, Kind.TYPE, 0); stop()
+            enterTypeParameters(t.tparams)
+          }
         case t: Decl.Type =>
           tpe(t.name, Kind.TYPE, 0); stop()
           enterTypeParameters(t.tparams)
         case t: Defn.Val =>
-          enterPatterns(t.pats, Kind.METHOD, Property.VAL.value); stop()
+          withScala3Owner {
+            enterPatterns(t.pats, Kind.METHOD, Property.VAL.value)
+            stop()
+          }
         case t: Decl.Val =>
           enterPatterns(t.pats, Kind.METHOD, Property.VAL.value); stop()
         case t: Defn.Var =>
-          enterPatterns(t.pats, Kind.METHOD, Property.VAR.value); stop()
+          withScala3Owner {
+            enterPatterns(t.pats, Kind.METHOD, Property.VAR.value); stop()
+          }
         case t: Decl.Var =>
-          enterPatterns(t.pats, Kind.METHOD, Property.VAR.value); stop()
+          withScala3Owner {
+            enterPatterns(t.pats, Kind.METHOD, Property.VAR.value); stop()
+          }
+        case t: Defn.Def =>
+          if (currentOwner.endsWith("/")) {
```

Comment:
Maybe add an additional method?
```suggestion
          if (isFileOwner) {
```

Reply:
Added `isPackageOwner`